### ISSUE TITLE
Allow text outside figure box

### ISSUE THREAD

**Issue by DasHammett** [Phlya/adjustText#91]
Hi, 
It would seem adjust_text forces the text to be inside the plot figure, but this behaviour have issues with what I would like to achieve.

I want to build a line chart with text at the end of each line identifying the characteristic.

I use the following code to achieve it, and it works fine:
```
text = []
for col in df.columns:                                                     
    text.append(plt.text(plt.xticks()[0][-1], df[col].iloc[-1],col))
```
This put the text outside of the figure plot, right beside each line. The problem arises when the lines end up too close and the text overlap.

I have been trying adjust_text with several options and the best I could achieve was with this
```
adjust_text(text,
            arrowprops=dict(arrowstyle="->", color='r', lw=0.5),           
            autoalign='', 
            only_move={'points':'y', 'text':'y'}) 
```
but this moves all labels inside the plot. In fact, all options I tried put them inside.

Is there an option/way to put them outside?

An example of what I want to achieve is here:
https://stackoverflow.com/questions/49237522/how-to-annotate-end-of-lines-using-python-and-matplotlib

Thanks

**Comment by Phlya** [comment]
No, at the moment there is no option - adjustText will always force texts to be inside the axes. You can try adding second axes just for texts. I'll see if it's easy to add an option to allow texts to extend bring beyond axes limits when I have time.

**Comment by Phlya** [comment]
There is now ensure_inside_axes argument, by default it's set to True.
